### PR TITLE
fix: store user id of element snapshot

### PR DESCRIPTION
**Pull request by marcoklein**

### CODE REVIEW

File: packages/impromat-api/test/graphql/services/element.service.test.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+          {
+            OR: [
+              { ownerId: userRequestId },
+              { visibility: PrismaElementVisibility.PUBLIC },
+            ],
+          },
+        ],
+      });
+
+      expect(createSnapshotElementMock.mock.calls[0][0].data).toEqual({
```
Comment (BreitbandModem) [inline packages/impromat-api/test/graphql/services/element.service.test.ts:77]: Best practice would be a single expect statement per test (maybe two).
here there are 5. If one of the expect statements fails, it's rather difficult to track down the source of the exact bug.

File: packages/impromat-api/test/graphql/services/element.service.test.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      expect(response).toEqual({
+        id: 'test-element',
+        visibility: 'PRIVATE',
+      });
+    });
+
+    it('should throw an error if the element is not existing', async () => {
+      // given
+      const userRequestId = 'test-user';
+      const updateInput: UpdateElementInput = {
```
Comment (BreitbandModem) [inline packages/impromat-api/test/graphql/services/element.service.test.ts:101]: if some of the mocks are required for all / many tests, then maybe they could be consolidated in the describe block?!

File: packages/impromat-api/test/graphql/services/element.service.test.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        .mockResolvedValueOnce({ ownerId: 'other-user' } as PrismaElement);
+      // when
+      try {
+        await service.updateElement(userRequestId, updateInput);
+        fail('Error expected.');
+      } catch (e) {
+        // then
+        expect(e).toBeInstanceOf(Error);
+        expect((e as Error).message).toContain('Write not permitted.');
+      }
```
Comment (BreitbandModem) [inline packages/impromat-api/test/graphql/services/element.service.test.ts:137]: I'm not familiar with the testing framework but usually they provide a test for errors. 
Something like  
`expect(service.updateElement(userRequestId, updateInput)).to.fail.with.message('Write not permitted')`  
That way you avoid having to write logic here in the test.

File: packages/impromat-api/test/graphql/services/element.service.test.ts (hunk 1 above)
Comment (BreitbandModem) [reply packages/impromat-api/test/graphql/services/element.service.test.ts:77]: e.g. add a describe block for `updating elements` with several tests:
1. repository is being called with the correct data
2. correct return value
3. transaction is called with correct data
4. transaction returns expected value
5. ...

File: packages/impromat-api/README.md [hunk 4]
```diff
@@ -51,6 +51,10 @@ Watch tests
 yarn test:watch
 ```
 
+## Service Template Test
```
Comment (BreitbandModem) [inline packages/impromat-api/README.md:54]: nice! what about the other test layers? ;P

**Review by BreitbandModem — COMMENTED** [review]
what do you mean you merged w/o review :O  

Tests are super readable and I wouldn't add any more abstraction layers etc.  
Just some cosmetic remarks and best practices to keep in mind :)

**Comment by marcoklein** [comment]
> what do you mean you merged w/o review :O
> 
> Tests are super readable and I wouldn't add any more abstraction layers etc. Just some cosmetic remarks and best practices to keep in mind :)

Thanks for your suggestions. I considered them in a new PR https://github.com/marcoklein/impromat/pull/212/files#diff-a80bee385430a8c89240e53e330740be3d8d7c53d82f8096128a1f61ec14bc80

**Pull request merged**